client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -2011,24 +2004,26 @@ impl NetworkBehaviour for Notifications {
 			return Poll::Ready(event)
 		}
 
-		// Poll for instructions from the peerset.
-		// Note that the peerset is a *best effort* crate, and we have to use defensive programming.
+		// Poll for instructions from the protocol controllers.
 		loop {
-			match futures::Stream::poll_next(Pin::new(&mut self.peerset), cx) {
-				Poll::Ready(Some(crate::peerset::Message::Accept(index))) => {
+			match futures::Stream::poll_next(Pin::new(&mut self.from_protocol_controllers), cx) {
```

Comment:
next().now_or_never()?

Reply:
`now_or_never()` is completely synchronous and knows nothing about async context, so wake-up won't be scheduled once the message is available.